bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -9285,6 +9314,42 @@ void setImageName(String name, long objId) throws TskCoreException {
 		}
 	}
 
+
+
+	/**
+	 * Updates the image's total size and sector size.This function may be used
+	 * to update the sizes after the image was created.
+	 *
+	 * Can only update the sizes if they were not set before. Will throw
+	 * TskCoreException if the values in the db are not 0 prior to this call.
+	 *
+	 * @param imgage     The image that needs to be updated
+	 * @param totalSize  The total size
+	 * @param sectorSize The sector size
+	 *
+	 * @throws TskCoreException Thrown if the values were not 0 in DB
+	 *
+	 */
+	void setImageSizes(Image image, long totalSize, long sectorSize) throws TskCoreException {
+
+		if (image.getSize() > 0 || image.getSsize() > 0) {
+			throw new TskCoreException(String.format("Error updating image sizes for object ID %d. Cannot update non-zero total size (%d) or sector size (%d)",image.getId(), image.getSize(), image.getSsize()));
+		}
```

Comment:
This method has a check-then-act race condition. Why is it necessary to enforce this restriction? Is seems like we are putting application/business logic in the data layer.

Reply:
Fair question. The only reason I have is that I assumed that size is immutable. Once the value is set, we should not allow the change. 
But seeing this is business logic,  I will remove the condition.